### PR TITLE
ODK Upgrade

### LINKED ISSUE #114: Make codebase compatible w/ latest ODK
## Overview
As of now, I'm getting errors if I run on `obolibrary/odkfull:` `latest` or `dev`. Right now we are stuck to `v1.4.3`.

## Sub-tasks
- [x] 1. Get working on `latest`
- [x] 2. Find out what version that container is, and update the GH action to use that version label rather than `latest`
  - [x] Update the reference in the GitHub action
  - [x] Update the reference in `run.sh`, or perhaps there it might indeed be best to use `dev` or `latest`

### Error details
Err: `KeyError: "['subject_label'] not in index"`
Happens when: creating `mondo_exactmatch_omimps.sssom.owl`

<details><summary>Details</summary>
<p>

```
sh run.sh make mondo_genes.robot.tsv; beep
sssom convert mondo_exactmatch_omimps.sssom.tsv -O owl -o mondo_exactmatch_omimps.sssom.owl
/usr/local/lib/python3.10/dist-packages/sssom/util.py:168: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df.replace("", np.nan, inplace=True)
Traceback (most recent call last):
  File "/usr/local/bin/sssom", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/dist-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/sssom/cli.py", line 161, in convert
    convert_file(input_path=input, output=output, output_format=output_format)
  File "/usr/local/lib/python3.10/dist-packages/sssom/io.py", line 50, in convert_file
    write_func(doc, output, serialisation=fileformat)  # type:ignore
  File "/usr/local/lib/python3.10/dist-packages/sssom/writers.py", line 162, in write_owl
    graph = to_owl_graph(msdf)
  File "/usr/local/lib/python3.10/dist-packages/sssom/writers.py", line 173, in to_owl_graph
    msdf.df = invert_mappings(
  File "/usr/local/lib/python3.10/dist-packages/sssom/util.py", line 1462, in invert_mappings
    inverted_df = inverted_df[df.columns]
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py", line 4108, in __getitem__
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py", line 6200, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py", line 6252, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['subject_label'] not in index"
make: *** [makefile:23: mondo_exactmatch_omimps.sssom.owl] Error 1
```

</p>
</details>

### LINKED ISSUE #127: Bug: `run.sh` not working: `ModuleNotFoundError`
## Overview
`run.sh` is currently not working, due to at least 1 package not being present in the ODK container:

`ModuleNotFoundError: No module named 'dotenv'`

## Possible solutions
For local use, could have a variation of `make all`, at least, which could have a `pip install` goal for this.

## Additional info
The GitHub action does not fail, but this doesn't work locally. I assume this is because the fixed version of ODK that the GH action is running off of does not have this same problem.

### PR DESCRIPTION
**Pull request by joeflack4**
resolves #114 
resolves #127

## Changes
- Upgrade to v1.5.3. Has a bugfix for SSSOM and includes dotenv package.

### CODE REVIEW

File: analyses/morbidmap-data-analysis/.ipynb_checkpoints/Analyze_morbidmap - v1-checkpoint.ipynb
Comment (joeflack4) [inline analyses/morbidmap-data-analysis/.ipynb_checkpoints/Analyze_morbidmap - v1-checkpoint.ipynb:1]: I will check later, but not sure yet why these notebooks are showing up in the diff. They should be on `main`, not in this PR.

File: makefile [hunk 1]
```diff
@@ -65,7 +65,7 @@ get-pmids:
 
 # SETUP / INSTALLATION -------------------------------------------------------------------------------------------------
 install:
-	pip install -r requirements-unlocked.txt
+	pip install -r requirements-unlocked.txt --user --break-system-packages
```
Comment (joeflack4) [inline makefile:68]: [Build passing now](https://github.com/monarch-initiative/omim/actions/runs/11674781775) after adding this.

Comment (twhetzel) [reply makefile:68]: Why is `--user --break-system-packages` needed? I thought that was just to add packages that aren't in ODK and ODK should now have everything this repo needs, right?

Comment (joeflack4) [reply makefile:68]: Here is [the rationale](https://github.com/monarch-initiative/omim/issues/114#issuecomment-2453575864). I'd rather not have to keep adding/removing `make install` every time I introduce a new package.

Comment (twhetzel) [reply makefile:68]: Hmm.... ok. Is it mentioned somewhere, other than a PR comment, that any new non-ODK package needed here does need an ODK issue to request it :)

Comment (joeflack4) [reply makefile:68]: I do have it in my 'rules', which I review every now and then, that I personally should open up a GH issue in such an event, but I wouldn't know where to write these things down; maybe some `mondo-ingest` developer documentation.

Also alternatively, if we wanted to make this `make install` thing more efficient, we could have it run a script that instead  iterates over each package in the `requirements.txt`, checks first if it is already installed globally, and then only do the `pip install PACKAGE --user --break-system-packages` only if it is not. Feels a little excessive, but in a way that is the most elegant approach. Right now it's not a major issue.

**Review by twhetzel — APPROVED** [review]
Approving now so you're not blocked later after the notebook files are moved out from this PR

**Pull request merged**